### PR TITLE
added DashedLine and CurvesAsSubmobjects example to docs

### PR DESCRIPTION
**Pull request by kolibril13**
I just had some minor things that could be improved, see in changelist

### CODE REVIEW

File: docs/source/examples.rst [hunk 1]
```diff
@@ -261,8 +262,9 @@ Animations
             path.set_points_as_corners([dot.get_center(), dot.get_center()])
             def update_path(path):
                 previous_path = path.copy()
-                previous_path.add_points_as_corners([dot.get_center()])
-                path.become(previous_path)
+                self.remove(previous_path)
+                path.add_points_as_corners([dot.get_center()])
+                self.add(path)
```
Comment (kolibril13) [inline docs/source/examples.rst:267]: this is way quicker then the previous solution

**Review by leotrs — APPROVED** [review]

File: manim/mobject/geometry.py [hunk 2]
```diff
@@ -710,6 +710,18 @@ def set_opacity(self, opacity, family=True):
 
 
 class DashedLine(Line):
+    """A dashed Line : - - - - - -
```
Comment (leotrs) [inline manim/mobject/geometry.py:713]: Suggestion (non-blocking): I'm not sure the visual aid here is necessary :sweat_smile: 
```suggestion
    """A dashed Line.
```

File: manim/mobject/geometry.py [hunk 3]
```diff
@@ -710,6 +710,18 @@ def set_opacity(self, opacity, family=True):
...
+
+    Examples
+    --------
+    .. manim:: DashedLineExample
+        :save_last_frame:
+
+        class DashedLineExample(Scene):
+            def construct(self):
+                l = DashedLine(config.frame_width/2*LEFT,4*RIGHT)
```
Comment (leotrs) [inline manim/mobject/geometry.py:722]:
```suggestion
                l = DashedLine(config.frame_width/2*LEFT, 4*RIGHT)
```

File: manim/mobject/geometry.py [hunk 4]
```diff
@@ -710,6 +710,18 @@ def set_opacity(self, opacity, family=True):
...
+                self.add(l)"""
```
Comment (leotrs) [inline manim/mobject/geometry.py:723]:
```suggestion
                self.add(l)
                
    """
```

File: manim/mobject/types/vectorized_mobject.py [hunk 5]
```diff
@@ -1388,6 +1388,21 @@ def set_location(self, new_loc):
 
 
 class CurvesAsSubmobjects(VGroup):
+    """Takes a  curve and converts elements to submobjects.
```
Comment (leotrs) [inline manim/mobject/types/vectorized_mobject.py:1391]:
```suggestion
    """Convert a curve's elements to submobjects.
```

File: manim/mobject/types/vectorized_mobject.py [hunk 6]
```diff
@@ -1388,6 +1388,21 @@ def set_location(self, new_loc):
...
+
+    Examples
+    --------
+    .. manim:: LineGradientExample
+        :save_last_frame:
+
+        class LineGradientExample(Scene):
+            def construct(self):
+                l = CurvedDoubleArrow(LEFT,RIGHT)
```
Comment (leotrs) [inline manim/mobject/types/vectorized_mobject.py:1400]:
```suggestion
                l = CurvedDoubleArrow(LEFT, RIGHT)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 7]
```diff
@@ -1388,6 +1388,21 @@ def set_location(self, new_loc):
...
+                new_func = CurvesAsSubmobjects(l, background_stroke_width =0)
```
Comment (leotrs) [inline manim/mobject/types/vectorized_mobject.py:1401]:
```suggestion
                new_func = CurvesAsSubmobjects(l, background_stroke_width=0)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 8]
```diff
@@ -1388,6 +1388,21 @@ def set_location(self, new_loc):
...
+                new_func.set_color_by_gradient(BLUE,RED)
+                self.add(new_func)
```
Comment (leotrs) [inline manim/mobject/types/vectorized_mobject.py:1403]:
```suggestion
                self.add(new_func)
                
```

File: manim/mobject/types/vectorized_mobject.py [hunk 9]
```diff
@@ -1388,6 +1388,21 @@ def set_location(self, new_loc):
...
+    Examples
+    --------
+    .. manim:: LineGradientExample
+        :save_last_frame:
+
+        class LineGradientExample(Scene):
+            def construct(self):
+                l = CurvedDoubleArrow(LEFT,RIGHT)
+                new_func = CurvesAsSubmobjects(l, background_stroke_width =0)
+                new_func.set_color_by_gradient(BLUE,RED)
```
Comment (leotrs) [inline manim/mobject/types/vectorized_mobject.py:1402]:
```suggestion
                new_func.set_color_by_gradient(BLUE, RED)
```

**Review by leotrs — CHANGES_REQUESTED** [review]

**Comment by leotrs** [comment]
@kolibril13 please remember to branch from your own fork :slightly_smiling_face:

File: manim/mobject/geometry.py (hunk 2 above)
Comment (kolibril13) [reply manim/mobject/geometry.py:713]: Ok, I just thought this would be an easter egg, but I don't mind removing it.

**Comment by kolibril13** [comment]
> @kolibril13 please remember to branch from your own fork

Thanks for your review!
Yess, I will keep this in mind.

File: docs/source/examples.rst (hunk 1 above)
Comment (kolibril13) [inline docs/source/examples.rst:267]:
```suggestion
                previous_path.add_points_as_corners([dot.get_center()])
                path.become(previous_path) 
```

File: docs/source/examples.rst [hunk 10]
```diff
@@ -262,7 +263,7 @@ Animations
             def update_path(path):
                 previous_path = path.copy()
                 previous_path.add_points_as_corners([dot.get_center()])
-                path.become(previous_path)
+                path.become(previous_path)
```
Comment (eulertour) [inline docs/source/examples.rst:266]: ?

Comment (kolibril13) [inline docs/source/examples.rst:266]:
```suggestion
                path.become(previous_path)
```

**Review by behackl — CHANGES_REQUESTED** [review]
The proposed changes look good to me. Please update the pull request title in order to reflect the actual changes from this PR (alternatively, prepare a one-line summary which can be included in the changelog).

**Comment by kolibril13** [comment]
Done! :)

**Review by leotrs — APPROVED** [review]

**Comment by behackl** [comment]
> Done! :)
> I think we don't have to include into the changelog what examples were added to the docs, do we?

My current policy is to include *all* merged PRs in the changelog (mainly in order to avoid the question whether or not a particular change is "worthy" of being included in the changelog) -- besides, having a complete list of merged PRs somewhere is nice.

File: manim/mobject/geometry.py [hunk 11]
```diff
@@ -710,6 +710,20 @@ def set_opacity(self, opacity, family=True):
...
+
+    Examples
+    --------
+    .. manim:: DashedLineExample
+        :save_last_frame:
+
+        class DashedLineExample(Scene):
+            def construct(self):
+                l = DashedLine(config.frame_width/2*LEFT, 4*RIGHT)
+                self.add(l)
```
Comment (behackl) [inline manim/mobject/geometry.py:723]: https://www.python.org/dev/peps/pep-0008/#names-to-avoid -- `l` should be avoided as a variable name.

```suggestion
                dashed_line = DashedLine(config.frame_width/2*LEFT, 4*RIGHT)
                self.add(dashed_line)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 12]
```diff
@@ -1388,6 +1388,22 @@ def set_location(self, new_loc):
...
+
+    Examples
+    --------
+    .. manim:: LineGradientExample
+        :save_last_frame:
+
+        class LineGradientExample(Scene):
+            def construct(self):
+                l = CurvedDoubleArrow(LEFT, RIGHT)
+                new_func = CurvesAsSubmobjects(l, background_stroke_width=0)
```
Comment (behackl) [inline manim/mobject/types/vectorized_mobject.py:1401]: Same comment as above, `l` should be avoided as a variable name.

Also, the example confuses me (or maybe I don't understand what `CurvesAsSubmobjects` is supposed to do: why are the arrow tips not included in the rendered image? (This is why I didn't suggest changing `l` to `arrow` or so.)

**Review by behackl — CHANGES_REQUESTED** [review]
Sorry for taking so long to look at this again, it was more busy than usual for the past few days. Two minor comments and a general question about `CurvesAsSubmobjects`.

File: manim/mobject/types/vectorized_mobject.py (hunk 12 above)
Comment (kolibril13) [reply manim/mobject/types/vectorized_mobject.py:1401]: image
Oh, you are right, the CurvedDoubleArrow loses its tips when using CurvesAsSubmobjects.
I will change this to a parametric function

File: manim/mobject/types/vectorized_mobject.py [hunk 13]
```diff
@@ -1388,6 +1388,22 @@ def set_location(self, new_loc):
...
+                new_func.set_color_by_gradient(BLUE, RED)
+                self.add(new_func)
```
Comment (kolibril13) [inline manim/mobject/types/vectorized_mobject.py:1403]:
```suggestion
          curve = ParametricFunction(lambda t: [t, np.sin(t), 0], -PI, t_max=PI,stroke_width=10)
          new_curve = CurvesAsSubmobjects(curve)
          new_curve.set_color_by_gradient(BLUE, RED)
          self.add(new_curve.shift(UP), curve)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 14]
```diff
@@ -1388,6 +1388,22 @@ def set_location(self, new_loc):
...
+    """Convert a curve's elements to submobjects.
+
+    Examples
+    --------
+    .. manim:: LineGradientExample
+        :save_last_frame:
+
+        class LineGradientExample(Scene):
+            def construct(self):
+          curve = ParametricFunction(lambda t: [t, np.sin(t), 0], -PI, t_max=PI,stroke_width=10)
```
Comment (kolibril13) [inline manim/mobject/types/vectorized_mobject.py:1400]:
```suggestion
          curve = ParametricFunction(lambda t: [t, np.sin(t), 0], t_min = -PI, t_max=PI,stroke_width=10)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 15]
```diff
@@ -1388,6 +1388,22 @@ def set_location(self, new_loc):
...
+    --------
+    .. manim:: LineGradientExample
+        :save_last_frame:
+
+        class LineGradientExample(Scene):
+            def construct(self):
+          curve = ParametricFunction(lambda t: [t, np.sin(t), 0], t_min = -PI, t_max=PI,stroke_width=10)
+          new_curve = CurvesAsSubmobjects(curve)
+          new_curve.set_color_by_gradient(BLUE, RED)
+          self.add(new_curve.shift(UP), curve)
```
Comment (kolibril13) [inline manim/mobject/types/vectorized_mobject.py:1403]:
```suggestion
                                curve = ParametricFunction(lambda t: [t, np.sin(t), 0], t_min = -PI, t_max=PI,stroke_width=10)
                                new_curve = CurvesAsSubmobjects(curve)
                                new_curve.set_color_by_gradient(BLUE, RED)
self.add(new_curve.shift(UP), curve)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 16]
```diff
@@ -1388,6 +1388,22 @@ def set_location(self, new_loc):
...
+    --------
+    .. manim:: LineGradientExample
+        :save_last_frame:
+
+        class LineGradientExample(Scene):
+            def construct(self):
+                                curve = ParametricFunction(lambda t: [t, np.sin(t), 0], t_min = -PI, t_max=PI,stroke_width=10)
+                                new_curve = CurvesAsSubmobjects(curve)
+                                new_curve.set_color_by_gradient(BLUE, RED)
+self.add(new_curve.shift(UP), curve)
```
Comment (kolibril13) [inline manim/mobject/types/vectorized_mobject.py:1403]:
```suggestion
            curve = ParametricFunction(lambda t: [t, np.sin(t), 0], t_min = -PI, t_max=PI,stroke_width=10)
            new_curve = CurvesAsSubmobjects(curve)
            new_curve.set_color_by_gradient(BLUE, RED)
            self.add(new_curve.shift(UP), curve)
```

**Review by behackl — APPROVED** [review]
LGTM, thank you for your work.

**Pull request merged**